Kja/controller

ktrops opened the pull request.

ktrops committed:
added controller tests for creating an instance of ShippingRate

ktrops committed:
functionality for creating a ShippingRate insance and rendering json

ktrops committed:
refactoring

ktrops committed:
tests

catchingash commented on app/controllers/shipping_api_controller.rb:
```diff
@@ -1,8 +1,14 @@
+      render json: estimates
+    else
+      render json: @shipping_rate.errors
+    end
```
We probably want to add status codes to these.

catchingash commented on app/controllers/shipping_api_controller.rb:
```diff
@@ -17,20 +23,41 @@ def create_log
-  def set_destination
+  def package
+    weight = package_params[:weight].to_i
+    dimensions = package_params[:dimensions].map{|d| d.to_i}
```
Is `to_i` necessary on weight and dimensions? Or can the gem handle these as strings?

ktrops replied:
Yes, they have to be integers. The api can't handle strings

catchingash commented on app/controllers/shipping_api_controller.rb:
```diff
@@ -17,20 +23,41 @@ def create_log
   def log_params
     params.require(:log).permit(:customer, :order_id, :service, :cost, :origin, :destination)
   end
+>>>>>>> e7401120388bcc98350f8dba9e38603eff5fb5e1
```
You'll want to handle the merge conflict here.

catchingash commented on spec/controllers/shipping_api_controller_spec.rb:
```diff
@@ -24,13 +26,39 @@
+        #   VCR.use_cassette("/rates", record: :new_episodes) do
+        #     post :rates, params
+        #   end
+        # end
```
Looks like we can delete these lines?

catchingash commented on spec/controllers/shipping_api_controller_spec.rb:
```diff
@@ -24,13 +26,39 @@
+        it "includes only service, price, and delivery_date" do
+          object.each do |instance|
+            expect(instance.keys).to eq(["service", "price", "delivery_date"])
+          end
```
Would you be willing to add another line to this one that checks to make sure that these are the ONLY keys included?

ktrops committed:
testing

catchingash commented on app/controllers/shipping_api_controller.rb:
```diff
@@ -1,8 +1,15 @@
+      estimates = @shipping_rate.ups_rates + @shipping_rate.usps_rates
+      render json: estimates
+    else
+      render json: @shipping_rate.errors
```
We still need status codes on these, don't we?

catchingash committed:
Added quick code change

Katie already updated these on her branch, but I want to merge the PR so I'm updating them on GH.
